dds_web/api/project.py

```diff
@@ -188,6 +188,123 @@ def post(self):
 
         return {"message": return_message}
 
+    @auth.login_required(role=["Unit Admin", "Unit Personnel"])
+    @logging_bind_request
+    @json_required
+    @handle_validation_errors
+    @handle_db_error
+    def patch(self):
+        """Partially update a the project status"""
+        # Get project ID, project and verify access
+        project_id = dds_web.utils.get_required_item(obj=flask.request.args, req="project")
+        project = dds_web.utils.collect_project(project_id=project_id)
+        dds_web.utils.verify_project_access(project=project)
+
+        # Get json input from request
+        json_input = flask.request.get_json(silent=True)  # Already checked by json_required
+
+        # Operation must be confirmed by the user - False by default
+        confirmed_operation = json_input.get("confirmed", False)
+        if not isinstance(confirmed_operation, bool):
+            raise DDSArgumentError(message="`confirmed` is a boolean value: True or False.")
+        if not confirmed_operation:
+            warning_message = "Operation must be confirmed before proceding."
+            # When not confirmed, return information about the project
+            project_info = ProjectInfo().get()
+            project_status = self.get()
+            json_returned = {
+                **project_info,
+                "project_status": project_status,
+                "warning": warning_message,
+                "default_unit_days": project.responsible_unit.days_in_expired,
+            }
+            return json_returned
+
+        # Cannot change project status if project is busy
+        if project.busy:
+            raise ProjectBusyError(
+                message=(
+                    f"The deadline for the project '{project_id}' is already in the process of being changed. "
+                    "Please try again later. \n\nIf you know that the project is not busy, contact support."
+                )
+            )
+
```

Comment:
Your test `test_extend_deadline_maxium_number_available_exceded` reminded me that we don't have a check for this prior to attempting the extension of the deadline. We need to check `project.times_expired` before trying to expire again.

Reply:
(And also before setting the project as busy)

Reply:
You mean do the check of times expired before the set busy?

Reply:
Yeah

Reply:
but if the check raises a DDSArgumentError. It will be caught by the last _except_ anyway and change the state to not busy again

Reply:
Yeah, but it's unnecessary for the code to go through all the following checks if it's anyway not possible to extend the deadline. 
For example if a user inputs a deadline that will not be approved, the user would currently potentially be changing the values and spending time on running the command multiple times, even though it won't matter since they can't change the deadline anyway. 

So actually we should put the `times_expired` check before the `confirmed_operation` part. 
1. check number of times_expired and return ProjectStatusError or something like that
2. check if operation confirmed 
3. continue with rest

Reply:
Plus the message that the user gets in the end is "Project availability limit: The maximun number of changes in data availability has been reached." which will confuse the users since they're not running `dds project status release` and they'll be thinking "I'm not trying to make the project available though, I'm just extending the deadline"

Reply:
Actually ignore that, the error message is fine. But the placement of the check should be changed to before `confirmed` check

Reply:
Or at least in the `try` but before all the other checks